src/parsemultipart.jl

```diff
@@ -0,0 +1,52 @@
+function parse_multipart_chunk!(chunk, d)
+    re = r"(\n){2,}|(\r){2,}|[\r\n]{4,}"
+    Parsers.exec(re, chunk) || return nothing
+    i, j = re.ovec
+    description = String(chunk[1:i])
+    content = Parsers.nextbytes(re, chunk)
+
+    v = match(r"Content-Disposition: form-data; name=\"(.*)\"; filename=\"(.*)\"[\r\n]+Content-Type: (\S*)", description)
+    if v !== nothing
+        push!(d["name"], v[1])
+        push!(d["filename"], v[2])
+        push!(d["contenttype"], v[3])
+    else
+        v = match(r"Content-Disposition: form-data; name=\"(.*)\"", description)
+        v == nothing && return
+        push!(d["name"], v[1])
+        push!(d["filename"], nothing)
+        push!(d["contenttype"], nothing)
+    end
+
+    re = r"(?:[\r\n]*)(?:-*)$"
+    Parsers.exec(re, content) && (content = content[1:re.ovec[1]])
+    push!(d["content"], content)
+end
+
+parse_multipart_body(body::AbstractArray{UInt8}, boundary) =
+    parse_multipart_body(String(body), boundary)
```

Comment:
Note that in 0.7 this is destructive; i.e. w/ `body` as a `Vector{UInt8}`, it will be turned into a `Vector{UInt8}(0)` after the String is created. Not saying that's necessarily bad, but if someone tried to access `body` again after calling this, it would not be what they expected. We should at least have docs that clearly spelled this behavior out.

Reply:
Agreed, at least the name should have a `!`, will try to run this code in 0.7 (I still haven't transitioned)

Reply:
What's the alternative? `String(copy(body))` ?